### PR TITLE
Update migration guide

### PR DESCRIPTION
**Pull request by adpi2**
Fix more links

Port latest changes:
- https://github.com/scalacenter/scala-3-migration-guide/pull/203 by @neko-kai
- https://github.com/scalacenter/scala-3-migration-guide/pull/204 by @unkarjedy

### CODE REVIEW

File: _overviews/scala3-migration/plugin-intro.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+title: Compiler Plugins
+type: chapter
+description: This section shows how to migrate from using Scala 2 compiler plugins
+num: 24
+previous-page: options-new
+next-page: plugin-kind-projector
+---
+
+Scala 3 includes some features that were previsouly provided by a compiler plugin.
```
Comment (b-studios) [inline _overviews/scala3-migration/plugin-intro.md:10]:
```suggestion
Scala 3 includes some features that were previously provided by a compiler plugin.
```

File: _overviews/scala3-migration/plugin-kind-projector.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
+---
+title: Kind Projector Migration
+type: section
+description: This section shows how to migrate from the kind-projector plugin to Scala 3 kind-projector syntax
+num: 25
+previous-page: plugin-intro
+next-page:
+---
+
+In the future, Scala 3 will use the `_` underscore symbol for placeholders in type lambdas---just as the underscore is currently used for placeholders in ordinary lambdas.
```
Comment (b-studios) [inline _overviews/scala3-migration/plugin-kind-projector.md:10]:
```suggestion
In the future, Scala 3 will use the `_` underscore symbol for placeholders in type lambdas---just as the underscore is currently used for placeholders in (ordinary) term-level lambdas.
```

File: _overviews/scala3-migration/plugin-kind-projector.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+```scala
+Tuple2[_, Double]        // equivalent to: type R[A] = Tuple2[A, Double]
+Either[Int, +_]          // equivalent to: type R[+A] = Either[Int, A]
+Function2[-_, Long, +_]  // equivalent to: type R[-A, +B] = Function2[A, Long, B]
+```
+
+## Compiling Existing Code
+
+Even without migrating to Underscore type lambdas, you will likely be able to compile most of it with Scala 3 without changes.
```
Comment (b-studios) [inline _overviews/scala3-migration/plugin-kind-projector.md:65]:
```suggestion
Even without migrating to Underscore Type Lambdas, you will likely be able to compile most of it with Scala 3 without changes.
```

At least this is how you capitalize it elsewhere :)

**Review by b-studios — APPROVED** [review]
LGTM

**Comment by b-studios** [comment]
Wait, the tests fail because a link is broken:

```
- ./_site/scala3/guides/migration/tutorial-prerequisites.html
  *  internally linking to plugin-kind-projector, which does not exist (line 0)
```

File: _overviews/scala3-migration/plugin-kind-projector.md (hunk 2 above)
Comment (neko-kai) [reply _overviews/scala3-migration/plugin-kind-projector.md:10]: I specifically avoided writing 'term-level' or 'value-level' 😄

File: _overviews/scala3-migration/plugin-kind-projector.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+type: section
+description: This section shows how to migrate from the kind-projector plugin to Scala 3 kind-projector syntax
+num: 25
+previous-page: plugin-intro
+next-page:
+---
+
+In the future, Scala 3 will use the `_` underscore symbol for placeholders in type lambdas---just as the underscore is currently used for placeholders in (ordinary) term-level lambdas.
+
+```scala
```
Comment (neko-kai) [inline _overviews/scala3-migration/plugin-kind-projector.md:12]: This code block is empty, might be a typo

File: _overviews/scala3-migration/plugin-kind-projector.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+To enable it, add the compiler flags `-Xsource:3 -P:kind-projector:underscore-placeholders` to your build.
+As in Scala 3, this will disable usage of `_` as a wildcard, however, the flag `-Xsource:3` will allow you to replace it with the `?` symbol. 
+
+The following `sbt` configuration will set up the correct flags to cross-compile with new syntax:
+
+```scala
+ThisBuild / scalacOptions ++= {
+  CrossVersion.partialVersion(scalaVersion.value) match {
+    case Some((3, _)) => Seq("-Ykind-projector:underscores")
+    case Some((2, 13)) => Seq("-Xsource:3", "-P:kind-projector:underscore-placeholders"))
```
Comment (neko-kai) [inline _overviews/scala3-migration/plugin-kind-projector.md:28]: @adpi2 The options work in `2.12` as well, although I’m not sure if these guides cover 2.12.

**Comment by lrytz** [comment]
@b-studios in case this is ready, I put a link to it into the 2.13.6 release notes :)

**Comment by b-studios** [comment]
It looks good to me, there is just this one comment by @neko-kai about empty code fences.  @adpi2 could you briefly comment on this?

**Comment by neko-kai** [comment]
@b-studios I think it's acceptable to merge and remove the fence in a post-merge commit. It's unlikely there's an intention behind it, since it's missing in the original migration guide repository, so I see no reason to fret over it too much if there's a hurry for release notes inclusion...

**Comment by adpi2** [comment]
Should be ready to merge now.

**Comment by b-studios** [comment]
Thanks @adpi2 !

**Pull request merged**